source/operators/rekognition/start_custom_label_detection.py

```diff
@@ -0,0 +1,140 @@
+
+    valid_image_types = [".json"]
+    file_type = os.path.splitext(s3key)[1]
+    
+    # Image batch processing can be synchronous / asynchronous
+    if file_type in valid_image_types:
+        
+        # Read metadata and list of frames
+        chunk_details = json.loads(s3.get_object(Bucket=s3bucket, Key=s3key, )["Body"].read())
+        
+        chunk_result = []
+        print("Number of chunks are " + str(len(chunk_details)))
+        #separate async processing logic for fan out architecture
+        if(FAN_OUT=='true') :
+            count = 1
+            total = len(chunk_details['s3_original_frames_keys'])
+            print("Number of frames to process are: " + str(total))
+            for img_s3key in chunk_details['s3_original_frames_keys']:
+                #trigger async lambda
+                end = False
+                if(count==total) :
+                    print("reached the end")
+                    end = True
+                response = client.invoke(FunctionName=fanout_function,InvocationType='Event',Payload=json.dumps( {'chunk_details':chunk_details,'asset_id': asset_id,'operator_name': operator_name,'workflow_id': workflow_id,'projectVersionArn':projectVersionArn, 'bucket': s3bucket, 'img_s3key': img_s3key,'end': end}))
+                print('response on async lambda invoke: %s', response)
+                count = count + 1
+                if(count % TPS ==0) :
+                    time.sleep(1)
+                    print('slept for 1 second')
+            output_object.update_workflow_status("Complete")
+            output_object.add_workflow_metadata(AssetId=asset_id, WorkflowExecutionId=workflow_id)
+            print('completed output object')
+            return output_object.return_output_object()
+        else : 
+            #for img_s3key in chunk_details['s3_resized_frame_keys']:
+            for img_s3key in chunk_details['s3_original_frames_keys']:
+                # For each frame detect custom labels and save the results
+                response = detect_custom_labels(s3bucket, urllib.parse.unquote_plus(img_s3key))
+                frame_result = []
+                for i in response['CustomLabels']:
+                    if i['Confidence'] > MIN_CONFIDENCE:
+                        bbox = i['Geometry']['BoundingBox']
+                        frame_id, file_extension = os.path.splitext(os.path.basename(img_s3key))
+                        frame_result.append({'frame_id': frame_id[3:],
+                                    'Text': {
+                                        'BoundingBox': bbox
+                                    },
+                                    'Confidence': i['Confidence'],
+                                    'Name': i['Name'],
+                                    'Timestamp': chunk_details['timestamps'][frame_id]})
+                        print("frame result is " + str(frame_result))
+                if len(frame_result)>0: chunk_result+=frame_result
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`.

 [Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)